src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/AtomPub10ServiceDocumentFormatter.cs

```diff
@@ -237,18 +198,10 @@ private static void ReadInlineCategories(XmlReader reader, InlineCategoriesDocum
                         {
                             continue;
                         }
+
                         string val = reader.Value;
-                        if (!TryParseAttribute(name, ns, val, inlineCategories, version))
-                        {
-                            if (preserveAttributeExtensions)
-                            {
-                                inlineCategories.AttributeExtensions.Add(new XmlQualifiedName(reader.LocalName, reader.NamespaceURI), reader.Value);
-                            }
-                            else
-                            {
-                                SyndicationFeedFormatter.TraceSyndicationElementIgnoredOnRead(reader);
-                            }
-                        }
+                        Debug.Assert(!TryParseAttribute(name, ns, val, inlineCategories, version));
```

Comment:
Why did this become an assert rather than actually guarding the next line?

Reply:
For all of these, the return value is always false:

```csharp
protected internal virtual bool TryParseAttribute(string name, string ns, string value, string version) => false;
protected internal virtual bool TryParseElement(XmlReader reader, string version) => false;
```